www/viewgame

```diff
@@ -1326,9 +1337,230 @@ function updateUnwishList(id, stat)
         . " &nbsp; <span id=\"ckUnwishlistSaveMsg\" class=\"xmlstatmsg\">"
         . "</span><br></div></div>";
 
-    ?>
-            <?php
+ 
+//-----------PHP for the Game Length Panel----------------------
 
+
+    // Check the database to see if the user has a "timevote" stored for this game.
+    // If one exists, return the stored time in minutes. If not, return 0.
+    $mystoredtime=0;
+
+    $result = mysqli_execute_query($db, "select timevote from playertimes where gameid = ? and userid = ?", [$id, $curuser]);
+    if (!$result) throw new Exception("Error: " . mysqli_error($db));
+        while ($row = mysqli_fetch_assoc($result)) {     // There should only be one row, at most
+            $mystoredtime = $row['timevote']; 
+        }
+    // At this point $mystoredtime should reflect the player's actual stored time for this game
+    $my_time_as_text = "";
+    $time_announcement = ""; // The string shown in the Game Length panel
+    if ($mystoredtime >= 1) {
+        $my_time_as_text=convertTimeToText($mystoredtime);
+        $time_announcement = "Your vote: " . $my_time_as_text;
+    }
+   
+
+
+//-----------------------Game Length Panel Begins----------------------------
+        echo "<div class=indented>";
+        echo "<div id='gameLengthPanel' class='viewgame__playedSection'>"
+        . "<span id=\"gameLengthCaption\"><b>Game length:</b></span>";
+
+        echo "<div id='submitTimeSection'>"
+        . "<span id='howLong'>Vote on long it takes to finish this game.</span><br>"
+        . "<details><summary>Tips</summary>"
+        . "<ul>"
+        . "<li>Vote only if you've played the game.</li>"
+        . "<li>Vote on how long it would take to get to one final ending without hints or a walkthrough.</li>"
+        . "<li>Time spent away from the game doesn't count.</li>"
+        . "<li>Times don't need to be exact. Good-faith estimates are fine."
+        . "<ul></details>"
+        . "<p><label for='hours'>Hours:</label> "
+        . "<input type='number' name='hours' id='hours' min=0 max=200> "
+        . "<label for='minutes'>Minutes:</label> "
+        . "<input type='number' name='minutes' id='minutes' min=0 max=59></p>"
+        . "<button id='submitTimeButton' class='fancy-button' type='submit'>Submit Time</button>"
+	    . "</div>";
+
+        echo "<div id='removeTimeSection'>"
+        . "<span id='yourTime'>" . $time_announcement . "</span><br>"
+        . "<button id='removeTimeButton' class='fancy-button'>Remove Time</button>"
+        . "</div>";
+
+        echo " <span id=\"timeSaveMsg\" class=xmlstatmsg aria-live='polite'></span><br>";
+
+	    echo "<div id='timeSavedSection'>"
+        . "<p id='timeSubmittedMessage'>Your time has been submitted.</p>"
+	    . "</div>";
+
+
+// ---------------------Javascript for the Game Length Panel---------------------
+?>
+
+<script type="text/JavaScript" nonce="<?php global $nonce; echo $nonce; ?>">
```

Comment:
no need for `type="text/javascript"` - it's the default. And all the code can be in a single <script> tag, no need to create several of them, at least in this case.

Also consider moving all JS for this feature into a separate file.

Reply:
Yes, I'm planning to get rid of the extra script tags once everything is more or less in place. I was running into problems where I would move a chunk of code and it would break things.